docs/development/core/server/kibana-plugin-server.clusterclient.callwithrequest.md

```diff
@@ -0,0 +1,13 @@
+<!-- Do not edit this file. It is automatically generated by API Documenter. -->
+
+[Home](./index.md) &gt; [kibana-plugin-server](./kibana-plugin-server.md) &gt; [ClusterClient](./kibana-plugin-server.clusterclient.md) &gt; [callWithRequest](./kibana-plugin-server.clusterclient.callwithrequest.md)
+
+## ClusterClient.callWithRequest property
+
+Calls specified `endpoint` with provided `clientParams` on behalf of the user initiated request to the Kibana server (via HTTP request headers).
+
+<b>Signature:</b>
+
+```typescript
+callWithRequest: (request: Request | KibanaRequest<unknown, unknown, unknown> | FakeRequest, endpoint: string, clientParams?: Record<string, unknown>, options?: CallAPIOptions | undefined) => Promise<any>;

```

Comment:
Any idea why the type for the `request` param in the docs is different than the code? In the code I see `request: KibanaRequest | LegacyRequest | FakeRequest`

Reply:
yes, because type alias is resolved to `Hapi.Request` eventually. I changes declaration to
```ts
interface LegacyRequest extends Request {}
```
to keep `LegacyRequest` as separate type

Reply:
Got it, thanks. I'm confused by how api-extractor/api-documenter resolves types then, but that's out of the scope of this PR. It seems to do it in some cases and not in others.